qadence/models/qnn_from_config.py

```diff
@@ -0,0 +1,743 @@
+    else:
+        raise ValueError(
+            f"Invalid ansatz strategy {config.ansatz_strategy} provided. Only 'digital', 'sdaqc', \
+                and 'rydberg' allowed"
+        )
+
+
+def _create_ansatz(
+    register: int | Register,
+    config: AnsatzConfig,
+) -> AbstractBlock:
+    """
+    Create the ansatz based on the configuration.
+
+    Args:
+        register (int | Register): Number of qubits or a register object.
+        config (AnsatzConfig): Configuration for the ansatz.
+
+    Returns:
+        AbstractBlock: The ansatz block.
+
+    Raises:
+        NotImplementedError: If the ansatz type is not implemented.
+    """
+    num_qubits = register if isinstance(register, int) else register.n_qubits
+
+    if config.ansatz_type == "iia":
+        return _create_iia(num_qubits=num_qubits, config=config)
+    elif config.ansatz_type == "hea":
+        return _create_hea_ansatz(register=register, config=config)
+    else:
+        raise NotImplementedError(
+            f"Ansatz of type {config.ansatz_type} not implemented yet. Only 'hea' and\
+                'iia' available."
+        )
+
+
+def _interleave_ansatz_in_fm(
+    register: int | Register,
+    fm_blocks: list[AbstractBlock],
+    ansatz_config: AnsatzConfig,
+) -> ChainBlock:
+    """
+    Interleave the ansatz layers in between the feature map layers.
+
+    Args:
+        register (int | Register): Number of qubits or a register object.
+        fm_blocks (list[AbstractBlock]): List of feature map blocks.
+        ansatz_config (AnsatzConfig): Ansatz configuration.
+
+    Returns:
+        ChainBlock: A block containing feature map layers with interleaved ansatz layers.
+    """
+    full_fm = []
+    for idx, block in enumerate(fm_blocks):
+        full_fm.append(block)
+        if idx + 1 < len(fm_blocks):
+            full_fm.append(_ansatz_layer(register, ansatz_config, idx))
+
+    return chain(*full_fm)
+
+
+def _get_observable_shifting_and_scaling(config: ObservableConfig) -> tuple[float, float]:
+    """
+    Get the observable shifting and scaling factors.
+
+    Args:
+        config (ObservableConfig): Observable configuration.
+
+    Returns:
+        tuple[float, float]: The observable shifting and scaling factors.
+    """
+    if config.output_range is None:
+        shift = 0.0
+        scale = 1.0
+
+    else:
+        if config.detuning is N:
+            shift = config.output_range[0]
+            scale = config.output_range[1] - config.output_range[0]
+        else:
+            shift = 0.5 * (config.output_range[0] + config.output_range[1])
+            scale = 0.5 * (config.output_range[1] - config.output_range[0])
+
+    return shift, scale
+
+
+def _global_identity(register: int | Register) -> KronBlock:
+    """Create a global identity block."""
+    return kron(
+        *[I(i) for i in range(register if isinstance(register, int) else register.n_qubits)]
+    )
+
+
+def _create_observable(
```

Comment:
@dominikandreasseitz this function here creates the required observable. Has the global identity to help it shift and then the appropriate scaling.

If no range is provided, then no transformation occurs.